twill-core/src/main/java/org/apache/twill/internal/TwillContainerLauncher.java

```diff
@@ -84,22 +84,29 @@ public TwillContainerLauncher(RuntimeSpecification runtimeSpec, ContainerInfo co
    * @param classPath The class path to load classes for the container.
    * @return instance of {@link TwillContainerController} to control the container run.
    */
-  public TwillContainerController start(RunId runId, int instanceId, Class<?> mainClass, String classPath) {
+  public TwillContainerController start(RunId runId, int instanceId, Class<?> mainClass, String classPath,
+                                        Location logLevelLocation) {
```

Comment:
Is `logLevelLocation` nullable? I see you check that in line 104. If that's the case, annotate this parameter with `@Nullable`. Also, please update the javadoc accordingly.